.github/workflows/build-and-publish.yml

```diff
@@ -118,50 +118,18 @@ jobs:
     name: Run code analysis
     # This job will not run until build_and_test is complete and passes.
     needs: build_and_test
+    # Call the reusable Snyk workflow and provide it the Snyk token through inheriting.
+    uses: ./.github/workflows/snyk-scan.yml
```

Comment:
So this re-runs the scan here in publish?  We weren't able to pull the persisted results or whatever from builld-and-test workflow?

Reply:
I don't know yet how the Snyk results are persisted, or if they're accessible between different workflow runs. I'll look more into how I could do that. It currently runs a scan in build-and-test and a separate scan in build-and-publish.

Another option I though might work would be to make build and test reusable. But that wouldn't change the need to do two scans.

Reply:
Ok for now i think it should be fine.  We can work on improving it later.

Reply:
Snyk results are persisted to the vendor on the monitor step.

The `test` is how you gate on during a build/pull-request
The `monitor` takes the state and publishes it to snyk.io. This allows them to do the daily rescans based on new vulnerabilities. Typically I would recommend this to be done after deployment.

Reply:
yeah.  Though, doesn't monitor typically use the state generated by test?  This is what I'm talking about trying to persist like we do in circleci.

Reply:
After some testing and looking through more documentation, I've confirmed that Snyk monitor does not require anything from Snyk test to run.  With .NET projects, Snyk needs `dotnet restore` to be run in order to get the `project.assets.json` file.  Snyk test uses that file to look for issues when run.  Snyk monitor uses that file to report the project dependencies to the Snyk UI where they can be monitored.

https://docs.snyk.io/products/snyk-open-source/language-and-package-manager-support/snyk-for-.net